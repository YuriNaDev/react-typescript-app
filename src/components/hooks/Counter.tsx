import React, { useState } from 'react'
import { Typography, Box, Button, ButtonGroup } from '@material-ui/core'

const Counter: React.FC = () => {
	const [count, setCount] = useState<number>(0)

	const onIncrease = () => setCount(state => state + 1)
	const onDecrease = () => setCount(state => state - 1)

	return (
		<Box py={2}>
			<Typography variant="h4" gutterBottom>
				{count}
			</Typography>
			<Box>
				<ButtonGroup color="primary">
					<Button onClick={onIncrease}>+1</Button>
					<Button onClick={onDecrease}>-1</Button>
				</ButtonGroup>
			</Box>
		</Box>
	)
}

export default Counter
