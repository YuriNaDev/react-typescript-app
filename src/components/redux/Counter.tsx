import React, { useCallback } from 'react'
import { Typography, Box, Button, ButtonGroup, Grid } from '@material-ui/core'
import { useSelector, useDispatch } from 'react-redux'
import { increase, decrease } from '../../stores/counter'
import { RootState } from '../../stores'

const Counter: React.FC = () => {
	const count = useSelector((state: RootState) => state.counter.count)

	const dispatch = useDispatch()
	const onIncrease = useCallback(() => dispatch(increase()), [dispatch])
	const onDecrease = useCallback(() => dispatch(decrease()), [dispatch])

	return (
		<Grid item>
			<Typography variant="h4" gutterBottom>
				{count}
			</Typography>
			<Box>
				<ButtonGroup color="primary">
					<Button onClick={onIncrease}>+1</Button>
					<Button onClick={onDecrease}>-1</Button>
				</ButtonGroup>
			</Box>
		</Grid>
	)
}

export default Counter
