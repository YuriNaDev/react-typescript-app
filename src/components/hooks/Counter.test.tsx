import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Counter from './Counter'

describe('<Counter />', () => {
	test('matches snapshot', () => {
		const { container } = render(<Counter />)
		expect(container).toMatchSnapshot()
	})

	test('has a number and two buttons', () => {
		const { getByText } = render(<Counter />)
		getByText('0')
		getByText('+1')
		getByText('-1')
	})

	test('increase 1', () => {
		const { getByText } = render(<Counter />)

		const number = getByText('0')
		const increaseButton = getByText('+1')

		fireEvent.click(increaseButton)
		fireEvent.click(increaseButton)

		expect(number).toHaveTextContent('2')
	})

	test('decrease 1', () => {
		const { getByText } = render(<Counter />)

		const number = getByText('0')
		const decreaseButton = getByText('-1')

		fireEvent.click(decreaseButton)
		fireEvent.click(decreaseButton)

		expect(number).toHaveTextContent('-2')
	})
})
