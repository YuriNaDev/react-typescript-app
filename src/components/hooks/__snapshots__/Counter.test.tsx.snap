// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-2"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
    >
      0
    </h4>
    <div
      class="MuiBox-root MuiBox-root-33"
    >
      <div
        class="MuiButtonGroup-root"
        role="group"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            +1
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlinedHorizontal MuiButtonGroup-groupedOutlinedPrimary MuiButton-outlinedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            -1
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
